File: packages/blog/src/types/user.type.ts
/**
 * 用户角色信息
 */
export interface UserRole {
  id: number
  name: string
  code: string
}

/**
 * 用户资源信息 (根据 JSON 暂定为 any 数组，可根据后续业务修改)
 */
export interface UserResource {
  // 根据实际业务需求补充字段
  [key: string]: any
}

/**
 * 用户信息主接口
 */
export interface UserInfo {
  id: number
  name: string
  age: number
  phone: string
  email: string
  /** 密码哈希值 */
  password: string
  /** 生日，格式: YYYY-MM-DD */
  birth_day: string
  /** 创建时间，格式: ISO 8601 */
  create_time: string
  /** 状态: 1-启用, 0-禁用 (根据实际业务定义) */
  status: number
  /** 性别: 0-未知/保密, 1-男, 2-女 (根据实际业务定义) */
  sex: number
  /** 用户所属角色列表 */
  roles: UserRole[]
  /** 用户拥有的资源权限 */
  resources: UserResource[]
}

File: apps/api/src/type/user_type.py
from datetime import datetime, date
from typing import Any, List, Optional
from pydantic import BaseModel, Field
import uuid

class User(BaseModel):
    id: Optional[int] = Field(None, description="用户的唯一标识符，自增主键")
    name: str = Field(..., description="昵称")
    sex: int = Field(..., description="性别")
    age: int = Field(..., description="年龄")
    phone: str = Field(..., description="手机号")
    email: str = Field(..., description="邮箱")
    password: str = Field(..., description="密码")
    birth_day: date = Field(..., description="出生日期，格式为 YYYY-MM-DD")
    create_time: Optional[datetime] = Field(default_factory=datetime.now, description="创建时间")
    status: bool = Field(default=True, description="用户是否启用，默认启用")


class UserPublic(BaseModel):
    id: Optional[int] = Field(None, description="用户的唯一标识符，自增主键")
    name: str = Field(..., description="昵称")
    sex: int = Field(..., description="性别")
    age: int = Field(..., description="年龄")
    phone: str = Field(..., description="手机号")
    email: str = Field(..., description="邮箱")
    birth_day: date = Field(..., description="出生日期，格式为 YYYY-MM-DD")
    create_time: Optional[datetime] = Field(default_factory=datetime.now, description="创建时间")
    status: bool = Field(default=True, description="用户是否启用，默认启用")
    roles: list[dict[str, Any]] = Field(default_factory=list, description="用户角色")
    resources: list[dict[str, Any]] = Field(default_factory=list, description="用户资源权限")


class UserCreateResult(BaseModel):
    id: Optional[int] = Field(None, description="新建用户 ID")


class UserPage(BaseModel):
    list: List[UserPublic] = Field(default_factory=list, description="当前页用户列表")
    page: int = Field(..., description="当前页码")
    page_size: int = Field(..., description="每页数量")
    total: int = Field(..., description="总条数")
